NumberGuessingGame.java

```diff
@@ -0,0 +1,24 @@
+import java.util.Scanner;
+import java.util.Random;
+
+class Main {
+    public static void main(String[] args) {
+        Random rand = new Random();
+        int number = rand.nextInt(100) + 1;
+        Scanner scanner = new Scanner(System.in);
+
+        while (true) {
+            System.out.println("Guess a number between 1 and 100:");
+            int guess = scanner.nextInt();
+
+            if (guess < number) {
+                System.out.println("Too low!");
+            } else if (guess > number) {
+                System.out.println("Too high!");
+            } else {
+                System.out.println("Congratulations! You guessed correctly!");
+                break;
+            }
+        }
```

Comment:
Close the scanner to free up resources!

Reply:
Please re-consider the way of generating a number from 10 to 99.
The current version
  int targetNumber = rand.nextInt(99) + 1;
 will generate a number from 1 to 100